app/controllers/incomeSources/add/ForeignPropertyReportingMethodController.scala

```diff
@@ -0,0 +1,257 @@
+
+    if (isEnabled(IncomeSources)) {
+      itsaStatusService.hasMandatedOrVoluntaryStatusCurrentYear.flatMap {
+        case true =>
+
+          val maybeLatencyDetails = user.incomeSources.properties
+            .find(_.incomeSourceId.contains(id))
+            .flatMap(_.latencyDetails)
+
+          maybeLatencyDetails match {
+            case Some(details) =>
+              getForeignPropertyReportingMethodDetails(details).map {
+                case Some(viewModel) =>
+                  Ok(foreignPropertyReportingMethodView(
+                    addForeignPropertyReportingMethodForm = AddForeignPropertyReportingMethodForm.form,
+                    foreignPropertyReportingViewModel = viewModel,
+                    postAction = postAction,
+                    isAgent = isAgent
+                  ))
+                case _ => Redirect(redirectCall)
+              }
+            case _ =>
+              Logger("application").error(s"[ForeignPropertyReportingMethodController][handleRequest]: Failed to retrieve latency details")
+              Future.successful(Redirect(redirectCall))
+          }
+      }
+    } else Future(Ok(customNotFoundErrorView()))
+  }
+
+    private def handleSubmitRequest(id: String,
+                                    isAgent: Boolean,
+                                    itvcErrorHandler: ShowInternalServerError,
+                                    postAction: Call,
+                                    redirectCall: Call)
+                                   (implicit user: MtdItUser[_]): Future[Result] = {
+
+      if (isEnabled(IncomeSources)) {
+        AddForeignPropertyReportingMethodForm.form.bindFromRequest().fold(
+          hasErrors => {
+
+            val maybeLatencyDetails = user.incomeSources.properties
+              .find(_.incomeSourceId.contains(id))
+              .flatMap(_.latencyDetails)
+
+            val updatedForm = AddForeignPropertyReportingMethodForm.updateErrorMessagesWithValues(hasErrors)
+
+            maybeLatencyDetails match {
+              case Some(details) =>
+                getForeignPropertyReportingMethodDetails(details).map {
+                  case Some(viewModel) =>
+                    BadRequest(foreignPropertyReportingMethodView(
+                      addForeignPropertyReportingMethodForm = updatedForm,
+                      foreignPropertyReportingViewModel = viewModel,
+                      postAction = postAction,
+                      isAgent = isAgent
+                    ))
+                  case None => Redirect(redirectCall)
+                }
+              case _ =>
+                Logger("application").error(s"[ForeignPropertyReportingMethodController][handleRequest]: Failed to retrieve latency details")
+                Future.successful(Redirect(redirectCall))
+            }
+          },
+          valid => {
+            val taxYear1ReportingMethod = valid.taxYear1ReportingMethod
```

Comment:
incorrect handling of Option, you have to extra values from the Option and handle scenario when there is no value provide;  pls don't call .get directly, as well as .head when using List etc